jax/_src/interpreters/mlir.py

```diff
@@ -712,6 +712,11 @@ def register_lowering(prim: core.Primitive, rule: LoweringRule,
   if platform is None:
     _lowerings[prim] = rule
   else:
+    if not xb.is_known_platform(platform):
+      known_platforms = sorted(list(xb.known_platforms()))
+      warnings.warn(f"Registering an MLIR lowering rule for primitive {prim} "
```

Comment:
Why not error?

Reply:
I don't know whether the plugins are guaranteed to be registered before the lowerings, so I didn't want to make this a hard failure. Would it be ok to make this an error?

Reply:
I think they should be. @hawkinsp can we error here?

Reply:
I'd be tempted to make it an error also. If it's a concern that plugins are not registered before lowerings, then adding "warning spam" is no good either and reason enough not to do this.

Reply:
Turned it into an error.